presto-main/src/main/java/io/prestosql/sql/ExpressionUtils.java

```diff
@@ -330,21 +327,6 @@ else if (!seen.contains(expression)) {
         return result.build();
     }
 
-    public static Expression normalize(Expression expression)
-    {
-        if (expression instanceof NotExpression) {
-            NotExpression not = (NotExpression) expression;
-            if (not.getValue() instanceof ComparisonExpression && ((ComparisonExpression) not.getValue()).getOperator() != IS_DISTINCT_FROM) {
-                ComparisonExpression comparison = (ComparisonExpression) not.getValue();
-                return new ComparisonExpression(comparison.getOperator().negate(), comparison.getLeft(), comparison.getRight());
-            }
-            if (not.getValue() instanceof NotExpression) {
```

Comment:
I think we can have this one even for doubles

Reply:
See reply above: https://github.com/prestosql/presto/pull/4120/files/d0b09b28dabdbab6e432d69a666ba43bc8f9de8f#r444930782